### PR TITLE
💡 Created typescript definition file

### PR DESCRIPTION
**Pull request by Abourass**
Built off of the sample @vsviridov provided in issue #1. I cleaned that up and made it correct (to the best of my knowledge). Should provide adequate definition for those of us running svelte w/ typescript

### CODE REVIEW

**Comment by zerodevx** [comment]
Hi @Abourass - first of all, my apologies for the late reply, swarmed the last few days! Thanks so much for your contribution, **much** appreciated! I don't personally use typescript often and I'm not too familiar with best practices, so do bear with me. I've added some comments in the code and would love to hear your thoughts.

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// index.d.ts
+declare module "@zerodevx/svelte-toast";
+
+import type { SvelteComponent } from "svelte";
+
+export interface SvelteToastOptions {
```
Comment (zerodevx) [inline src/index.d.ts:6]: Do we add hints for default property values here?

File: src/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+export class SvelteToast extends SvelteComponent {
+  options: SvelteToastOptions;
+}
+
+declare namespace toast {
+  export function push(text: string, options?: SvelteToastOptions): number;
+  export function pop(id: number): void;
+  export function set(id: number, options: SvelteToastOptions): void;
+  export function _opts(options: SvelteToastOptions): SvelteToastOptions;
```
Comment (zerodevx) [inline src/index.d.ts:24]: `_opts` is private and not intended to be exposed in the API. Do we still declare the namespace here?

File: src/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+}
```
Comment (zerodevx) [inline src/index.d.ts:25]: Lastly, is it OK to run this file through [standardjs](https://standardjs.com/) for code formatting, or is it an anti-pattern?

File: src/index.d.ts (hunk 2 above)
Comment (Abourass) [reply src/index.d.ts:24]: I will remove that.

File: src/index.d.ts (hunk 3 above)
Comment (Abourass) [reply src/index.d.ts:25]: I can't say I've ever tried. I've stuck to the conventions I see when typescript generates its own definition files, so I'm inclined to say anti-pattern, but that's based purely on what I've inferred.

File: src/index.d.ts (hunk 1 above)
Comment (zerodevx) [reply src/index.d.ts:6]: Ah I see, if it's not too much trouble, could you add the JSDOC parts? (or perhaps the scaffolding and I can fill in the blanks - that will help a lot too.) Thanks for this!

Comment (zerodevx) [reply src/index.d.ts:6]: HI @Abourass, just a ping to see if you could scaffold the JSDOC parts into the typescript definition file too?

Comment (Abourass) [reply src/index.d.ts:20]: @zerodevx - Sorry about that, slipped my mind. Got that finished and added

**Comment by Abourass** [comment]
@zerodevx - I think I've resolved everything! Let me know if there's anything else

**Comment by zerodevx** [comment]
Thank for your contribution! 🎉

**Pull request merged**